include/deal.II/lac/utilities.h

```diff
@@ -147,6 +168,73 @@ namespace Utilities
       return diagonal[k-1] + f->l2_norm();
     }
 
+
+    template <typename OperatorType, typename VectorType>
+    void Chebyshev_filter(VectorType &x,
+                          const OperatorType &op,
+                          const unsigned int degree,
+                          const double a,
+                          const double b,
+                          const double a_L,
+                          VectorMemory<VectorType> &vector_memory)
+    {
+      Assert (degree > 0,
+              ExcNotImplemented());
+
+      const bool scale = !numbers::is_nan(a_L);
+      Assert (a < b,
+              ExcMessage("Lower bound of the unwanted spectrum should be smaller than the highest bound."));
+
+      Assert (a_L <= a || a_L >= b || !scale,
+              ExcMessage("Scaling point should be outside of the filtering range."));
+
+      // Setup auxiliary vectors:
+      typename VectorMemory<VectorType>::Pointer p_y(vector_memory);
+      typename VectorMemory<VectorType>::Pointer p_yn(vector_memory);
+
+      p_y->reinit(x);
+      p_yn->reinit(x);
+
+      // convinience to avoid pointers
+      VectorType &y   = *p_y;
+      VectorType &yn  = *p_yn;
+
+      // Below is an implementation of
+      // Algorithm 3.2 in Zhou et al, Journal of Computational Physics 274 (2014) 770-782
+      // with **a bugfix for sigma1**. Here is the original algorithm verbatim:
```

Comment:
I think this comment should actually be (also) in the documentation of the function.